documentation.ipynb

```diff
@@ -4,8 +4,335 @@
+    "    from requests.exceptions import HTTPError\n",
+    "    from httpx import HTTPStatusError\n",
+    "    import sqlite3\n",
+    "    import warnings\n",
+    "\n",
+    "warnings.filterwarnings('ignore')\n",
+    "\n",
+    "# Display to confirm successful import\n",
+    "print(\"Dependencies imported successfully.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 87,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# API keys for Cohere and Mistral AI models\n",
+    "# Path to the Milvus database URI for storing vector embeddings\n",
+    "# The name of the transformer model to use for generating embeddings\n",
+    "# Extract the directory path from the URI and create it if it doesn't exist\n",
+    "# Establish a connection to the Milvus database\n",
+    "# Initialize the embedding function with the transformer model\n",
+    "# Initialize a RecursiveUrlLoader to load web documents while preventing \n",
+    "# navigation outside the specified base URL\n",
+    "# Initialize a vector store with the specified collection name, embedding function, and connection settings\n",
+    "# Create a text splitter to divide the documents into chunks of specified size with overlap\n",
+    "# Create a vector store from the documents, dropping any existing collection with the same name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 80,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Config vaiables and required functions loaded\n"
+     ]
+    }
+   ],
+   "source": [
+    "COHERE_API_KEY='JPZrN72mRP2r9W5Dedo3Zk18VUS8voCHIPD7WLly'\n",
+    "MISTRAL_API_KEY='aCp49AkZAXNhTUgFziRT3Q8uO0GgWjBj'\n",
+    "MILVUS_URI = \"milvus/milvus_vector.db\"\n",
+    "MODEL_NAME = \"sentence-transformers/all-MiniLM-L6-v2\"\n",
+    "\n",
+    "def vector_store_check(uri):\n",
+    "    # Create the directory if it does not exist\n",
+    "    head = os.path.split(uri)\n",
+    "    os.makedirs(head[0], exist_ok=True)\n",
+    "    \n",
+    "    # Connect to the Milvus database\n",
+    "    connections.connect(\"default\",uri=uri)\n",
+    "\n",
+    "    # Return True if exists, False otherwise\n",
+    "    return utility.has_collection(\"IT_support\")\n",
+    "\n",
+    "def get_embedding_function():\n",
+    "    \"\"\"\n",
+    "    returns embedding function for the model\n",
+    "\n",
+    "    Returns:\n",
+    "        embedding function\n",
+    "    \"\"\"\n",
+    "    embedding_function = HuggingFaceEmbeddings(model_name=MODEL_NAME)\n",
+    "    return embedding_function\n",
+    "\n",
+    "def load_documents_from_web():\n",
+    "    \"\"\"\n",
+    "    Load the documents from the web and store the page contents\n",
+    "\n",
+    "    Returns:\n",
+    "        list: The documents loaded from the web\n",
+    "    \"\"\"\n",
+    "    loader = RecursiveUrlLoader(\n",
+    "        url=CORPUS_SOURCE,\n",
```

Comment:
please define `CORPUS_SOURCE` before this (probably in the first or second block where we are importing everything)